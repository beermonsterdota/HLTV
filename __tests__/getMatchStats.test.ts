import HLTV from '../src/index.js'
import { sleep } from '../src/utils.js'
import { expect, test } from 'vitest'

const FULL_INFO_NEWEST = 79928
const FULL_INFO = 62979
const RATING_1 = 22124
const SIX_PLAYERS = 80328

test('getMatchStats', async () => {
  await sleep(3000)
  expect(await HLTV.getMatchStats({ id: FULL_INFO })).toMatchSnapshot()
  await sleep(3000)
  expect(await HLTV.getMatchStats({ id: RATING_1 })).toMatchSnapshot()
  await sleep(3000)
  expect(await HLTV.getMatchStats({ id: FULL_INFO_NEWEST })).toMatchSnapshot()
  await sleep(3000)
  expect(await HLTV.getMatchStats({ id: SIX_PLAYERS })).toMatchSnapshot()
  await sleep(3000)
}, 500000)
