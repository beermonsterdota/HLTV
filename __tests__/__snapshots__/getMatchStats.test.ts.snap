// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getMatchStats 1`] = `
{
  "date": 1544901600000,
  "event": {
    "id": 4205,
    "name": "BLAST Pro Series Lisbon 2018",
  },
  "id": 62979,
  "mapStatIds": [
    79440,
    79446,
    79452,
  ],
  "matchId": 2329728,
  "overview": {
    "bestRating2": {
      "id": 7398,
      "name": "dupreeh",
      "value": 1.34,
    },
    "clutchesWon": {
      "team1": 0,
      "team2": 3,
    },
    "firstKills": {
      "team1": 41,
      "team2": 27,
    },
    "mostAWPKills": {
      "id": 7592,
      "name": "device",
      "value": 23,
    },
    "mostAssists": {
      "id": 7412,
      "name": "gla1ve",
      "value": 13,
    },
    "mostDamage": {
      "id": 7398,
      "name": "dupreeh",
      "value": 93.3,
    },
    "mostFirstKills": {
      "id": 7398,
      "name": "dupreeh",
      "value": 12,
    },
    "mostKills": {
      "id": 7398,
      "name": "dupreeh",
      "value": 58,
    },
  },
  "performanceOverview": {
    "team1": {},
    "team2": {},
  },
  "playerStats": {
    "team1": [
      {
        "ADR": 93.3,
        "KAST": 80.9,
        "assists": 13,
        "deaths": 47,
        "firstKillsDifference": 4,
        "flashAssists": 4,
        "hsKills": 24,
        "killDeathsDifference": 11,
        "kills": 58,
        "player": {
          "id": 7398,
          "name": "dupreeh",
        },
        "rating1": 1.34,
      },
      {
        "ADR": 79.4,
        "KAST": 75,
        "assists": 13,
        "deaths": 34,
        "firstKillsDifference": 5,
        "flashAssists": 5,
        "hsKills": 34,
        "killDeathsDifference": 18,
        "kills": 52,
        "player": {
          "id": 9032,
          "name": "Magisk",
        },
        "rating1": 1.27,
      },
      {
        "ADR": 76,
        "KAST": 69.1,
        "assists": 10,
        "deaths": 40,
        "firstKillsDifference": 4,
        "flashAssists": 3,
        "hsKills": 13,
        "killDeathsDifference": 8,
        "kills": 48,
        "player": {
          "id": 7592,
          "name": "device",
        },
        "rating1": 1.1,
      },
      {
        "ADR": 71.2,
        "KAST": 69.1,
        "assists": 15,
        "deaths": 40,
        "firstKillsDifference": 1,
        "flashAssists": 2,
        "hsKills": 13,
        "killDeathsDifference": 1,
        "kills": 41,
        "player": {
          "id": 7412,
          "name": "gla1ve",
        },
        "rating1": 1.01,
      },
      {
        "ADR": 72.1,
        "KAST": 67.6,
        "assists": 13,
        "deaths": 42,
        "firstKillsDifference": 0,
        "flashAssists": 2,
        "hsKills": 12,
        "killDeathsDifference": -1,
        "kills": 41,
        "player": {
          "id": 4954,
          "name": "Xyp9x",
        },
        "rating1": 0.96,
      },
    ],
    "team2": [
      {
        "ADR": 81.6,
        "KAST": 61.8,
        "assists": 7,
        "deaths": 44,
        "firstKillsDifference": -1,
        "flashAssists": 2,
        "hsKills": 28,
        "killDeathsDifference": 10,
        "kills": 54,
        "player": {
          "id": 7998,
          "name": "s1mple",
        },
        "rating1": 1.16,
      },
      {
        "ADR": 71.8,
        "KAST": 61.8,
        "assists": 9,
        "deaths": 48,
        "firstKillsDifference": -4,
        "flashAssists": 3,
        "hsKills": 27,
        "killDeathsDifference": -4,
        "kills": 44,
        "player": {
          "id": 7594,
          "name": "flamie",
        },
        "rating1": 0.96,
      },
      {
        "ADR": 70,
        "KAST": 52.9,
        "assists": 11,
        "deaths": 51,
        "firstKillsDifference": -1,
        "flashAssists": 3,
        "hsKills": 20,
        "killDeathsDifference": -10,
        "kills": 41,
        "player": {
          "id": 8918,
          "name": "electroNic",
        },
        "rating1": 0.93,
      },
      {
        "ADR": 58.7,
        "KAST": 69.1,
        "assists": 3,
        "deaths": 46,
        "firstKillsDifference": -4,
        "flashAssists": 0,
        "hsKills": 18,
        "killDeathsDifference": -4,
        "kills": 42,
        "player": {
          "id": 483,
          "name": "Edward",
        },
        "rating1": 0.86,
      },
      {
        "ADR": 49.8,
        "KAST": 55.9,
        "assists": 18,
        "deaths": 51,
        "firstKillsDifference": -4,
        "flashAssists": 7,
        "hsKills": 11,
        "killDeathsDifference": -29,
        "kills": 22,
        "player": {
          "id": 484,
          "name": "Zeus",
        },
        "rating1": 0.59,
      },
    ],
  },
  "result": {
    "team1MapsWon": 2,
    "team2MapsWon": 1,
  },
  "team1": {
    "id": 6665,
    "name": "Astralis",
  },
  "team2": {
    "id": 4608,
    "name": "Natus Vincere",
  },
}
`;

exports[`getMatchStats 2`] = `
{
  "date": 1370372400000,
  "event": {
    "id": 1147,
    "name": "RaidCall EMS One Summer 2013 Cup #4",
  },
  "id": 22124,
  "mapStatIds": [],
  "matchId": 2249884,
  "overview": {
    "bestRating1": {
      "id": 2469,
      "name": "cajunb",
      "value": 1.71,
    },
    "clutchesWon": {
      "team1": 1,
      "team2": 0,
    },
    "firstKills": {
      "team1": 4,
      "team2": 16,
    },
    "mostAWPKills": {
      "id": 7592,
      "name": "device",
      "value": 12,
    },
    "mostAssists": {
      "id": 7398,
      "name": "dupreeh",
      "value": 8,
    },
    "mostFirstKills": {
      "id": 7592,
      "name": "device",
      "value": 9,
    },
    "mostKills": {
      "id": 7592,
      "name": "device",
      "value": 22,
    },
  },
  "performanceOverview": {
    "team1": {},
    "team2": {},
  },
  "playerStats": {
    "team1": [
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 1,
        "deaths": 17,
        "firstKillsDifference": -2,
        "flashAssists": 0,
        "hsKills": 10,
        "killDeathsDifference": -2,
        "kills": 15,
        "player": {
          "id": 7679,
          "name": "paynt",
        },
        "rating1": 0.98,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 0,
        "deaths": 17,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 2,
        "killDeathsDifference": -7,
        "kills": 10,
        "player": {
          "id": 7612,
          "name": "jakem",
        },
        "rating1": 0.66,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 3,
        "deaths": 18,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 4,
        "killDeathsDifference": -12,
        "kills": 6,
        "player": {
          "id": 7294,
          "name": "mole",
        },
        "rating1": 0.39,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 0,
        "deaths": 20,
        "firstKillsDifference": -5,
        "flashAssists": 0,
        "hsKills": 3,
        "killDeathsDifference": -13,
        "kills": 7,
        "player": {
          "id": 7715,
          "name": "ralleh",
        },
        "rating1": 0.35,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 0,
        "deaths": 19,
        "firstKillsDifference": -3,
        "flashAssists": 0,
        "hsKills": 0,
        "killDeathsDifference": -16,
        "kills": 3,
        "player": {
          "id": 6318,
          "name": "wilzOOO",
        },
        "rating1": 0.17,
      },
    ],
    "team2": [
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 2,
        "deaths": 7,
        "firstKillsDifference": 4,
        "flashAssists": 0,
        "hsKills": 11,
        "killDeathsDifference": 13,
        "kills": 20,
        "player": {
          "id": 2469,
          "name": "cajunb",
        },
        "rating1": 1.71,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 3,
        "deaths": 10,
        "firstKillsDifference": 8,
        "flashAssists": 0,
        "hsKills": 8,
        "killDeathsDifference": 12,
        "kills": 22,
        "player": {
          "id": 7592,
          "name": "device",
        },
        "rating1": 1.65,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 4,
        "deaths": 9,
        "firstKillsDifference": 0,
        "flashAssists": 0,
        "hsKills": 6,
        "killDeathsDifference": 10,
        "kills": 19,
        "player": {
          "id": 7584,
          "name": "wantz",
        },
        "rating1": 1.5,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 8,
        "deaths": 8,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 11,
        "killDeathsDifference": 11,
        "kills": 19,
        "player": {
          "id": 7398,
          "name": "dupreeh",
        },
        "rating1": 1.46,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 4,
        "deaths": 7,
        "firstKillsDifference": 1,
        "flashAssists": 0,
        "hsKills": 4,
        "killDeathsDifference": 4,
        "kills": 11,
        "player": {
          "id": 7582,
          "name": "FeTiSh",
        },
        "rating1": 1.05,
      },
    ],
  },
  "result": {
    "team1MapsWon": 4,
    "team2MapsWon": 16,
  },
  "team1": {
    "id": 4822,
    "name": "Reason",
  },
  "team2": {
    "id": 4548,
    "name": "CPH Wolves",
  },
}
`;

exports[`getMatchStats 3`] = `
{
  "date": 1616065500000,
  "event": {
    "id": 5785,
    "name": "Snow Sweet Snow 3 Regionals",
  },
  "id": 79928,
  "mapStatIds": [
    116688,
    116692,
  ],
  "matchId": 2347210,
  "overview": {
    "bestRating2": {
      "id": 16410,
      "name": "Duplicate",
      "value": 1.43,
    },
    "clutchesWon": {
      "team1": 3,
      "team2": 1,
    },
    "firstKills": {
      "team1": 35,
      "team2": 17,
    },
    "mostAWPKills": {
      "id": 12379,
      "name": "Impulse",
      "value": 26,
    },
    "mostAssists": {
      "id": 16410,
      "name": "Duplicate",
      "value": 9,
    },
    "mostDamage": {
      "id": 16410,
      "name": "Duplicate",
      "value": 94.8,
    },
    "mostFirstKills": {
      "id": 13466,
      "name": "dennyslaw",
      "value": 14,
    },
    "mostKills": {
      "id": 16410,
      "name": "Duplicate",
      "value": 43,
    },
  },
  "performanceOverview": {
    "team1": {},
    "team2": {},
  },
  "playerStats": {
    "team1": [
      {
        "ADR": 94.8,
        "KAST": 75,
        "assists": 10,
        "deaths": 32,
        "firstKillsDifference": 1,
        "flashAssists": 1,
        "hsKills": 22,
        "killDeathsDifference": 11,
        "kills": 43,
        "player": {
          "id": 16410,
          "name": "Duplicate",
        },
        "rating1": 1.43,
      },
      {
        "ADR": 72.6,
        "KAST": 80.8,
        "assists": 8,
        "deaths": 29,
        "firstKillsDifference": 10,
        "flashAssists": 3,
        "hsKills": 23,
        "killDeathsDifference": 14,
        "kills": 43,
        "player": {
          "id": 13466,
          "name": "dennyslaw",
        },
        "rating1": 1.37,
      },
      {
        "ADR": 89.4,
        "KAST": 73.1,
        "assists": 12,
        "deaths": 33,
        "firstKillsDifference": 4,
        "flashAssists": 5,
        "hsKills": 29,
        "killDeathsDifference": 10,
        "kills": 43,
        "player": {
          "id": 11343,
          "name": "SHiPZ",
        },
        "rating1": 1.3,
      },
      {
        "ADR": 80.4,
        "KAST": 80.8,
        "assists": 16,
        "deaths": 27,
        "firstKillsDifference": 2,
        "flashAssists": 8,
        "hsKills": 17,
        "killDeathsDifference": 10,
        "kills": 37,
        "player": {
          "id": 17145,
          "name": "Rainwaker",
        },
        "rating1": 1.25,
      },
      {
        "ADR": 53.9,
        "KAST": 82.7,
        "assists": 8,
        "deaths": 26,
        "firstKillsDifference": 1,
        "flashAssists": 5,
        "hsKills": 13,
        "killDeathsDifference": 0,
        "kills": 26,
        "player": {
          "id": 18697,
          "name": "oxygeN",
        },
        "rating1": 1.02,
      },
    ],
    "team2": [
      {
        "ADR": 81.8,
        "KAST": 65.4,
        "assists": 9,
        "deaths": 33,
        "firstKillsDifference": -4,
        "flashAssists": 5,
        "hsKills": 10,
        "killDeathsDifference": 9,
        "kills": 42,
        "player": {
          "id": 12379,
          "name": "Impulse",
        },
        "rating1": 1.1,
      },
      {
        "ADR": 62.6,
        "KAST": 67.3,
        "assists": 7,
        "deaths": 37,
        "firstKillsDifference": 0,
        "flashAssists": 4,
        "hsKills": 16,
        "killDeathsDifference": -6,
        "kills": 31,
        "player": {
          "id": 4218,
          "name": "Dav",
        },
        "rating1": 0.88,
      },
      {
        "ADR": 67.7,
        "KAST": 61.5,
        "assists": 7,
        "deaths": 43,
        "firstKillsDifference": -9,
        "flashAssists": 0,
        "hsKills": 16,
        "killDeathsDifference": -13,
        "kills": 30,
        "player": {
          "id": 3732,
          "name": "DiMKE",
        },
        "rating1": 0.78,
      },
      {
        "ADR": 58.7,
        "KAST": 53.8,
        "assists": 10,
        "deaths": 41,
        "firstKillsDifference": -4,
        "flashAssists": 2,
        "hsKills": 10,
        "killDeathsDifference": -19,
        "kills": 22,
        "player": {
          "id": 3730,
          "name": "HOLMES",
        },
        "rating1": 0.7,
      },
      {
        "ADR": 53.1,
        "KAST": 57.7,
        "assists": 4,
        "deaths": 38,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 9,
        "killDeathsDifference": -16,
        "kills": 22,
        "player": {
          "id": 5351,
          "name": "aVN",
        },
        "rating1": 0.66,
      },
    ],
  },
  "result": {
    "team1MapsWon": 2,
    "team2MapsWon": 0,
  },
  "team1": {
    "id": 10386,
    "name": "SKADE",
  },
  "team2": {
    "id": 11001,
    "name": "ONYX",
  },
}
`;

exports[`getMatchStats 4`] = `
{
  "date": 1617377400000,
  "event": {
    "id": 5810,
    "name": "Funspark ULTI 2021 Europe Regional Series 1",
  },
  "id": 80328,
  "mapStatIds": [
    117608,
    117611,
    117622,
  ],
  "matchId": 2347664,
  "overview": {
    "bestRating2": {
      "id": 14932,
      "name": "faveN",
      "value": 1.29,
    },
    "clutchesWon": {
      "team1": 5,
      "team2": 4,
    },
    "firstKills": {
      "team1": 35,
      "team2": 48,
    },
    "mostAWPKills": {
      "id": 15370,
      "name": "slaxz-",
      "value": 33,
    },
    "mostAssists": {
      "id": 7511,
      "name": "denis",
      "value": 17,
    },
    "mostDamage": {
      "id": 7511,
      "name": "denis",
      "value": 90.2,
    },
    "mostFirstKills": {
      "id": 7511,
      "name": "denis",
      "value": 12,
    },
    "mostKills": {
      "id": 14932,
      "name": "faveN",
      "value": 70,
    },
  },
  "performanceOverview": {
    "team1": {},
    "team2": {},
  },
  "playerStats": {
    "team1": [
      {
        "ADR": 82.2,
        "KAST": 77.4,
        "assists": 8,
        "deaths": 35,
        "firstKillsDifference": 2,
        "flashAssists": 0,
        "hsKills": 19,
        "killDeathsDifference": 8,
        "kills": 43,
        "player": {
          "id": 18749,
          "name": "sausol",
        },
        "rating1": 1.25,
      },
      {
        "ADR": 86.1,
        "KAST": 70,
        "assists": 12,
        "deaths": 42,
        "firstKillsDifference": 3,
        "flashAssists": 1,
        "hsKills": 25,
        "killDeathsDifference": 5,
        "kills": 47,
        "player": {
          "id": 14007,
          "name": "tudsoN",
        },
        "rating1": 1.22,
      },
      {
        "ADR": 78.3,
        "KAST": 67.5,
        "assists": 11,
        "deaths": 59,
        "firstKillsDifference": 0,
        "flashAssists": 2,
        "hsKills": 24,
        "killDeathsDifference": -10,
        "kills": 49,
        "player": {
          "id": 11995,
          "name": "TMB",
        },
        "rating1": 0.99,
      },
      {
        "ADR": 69.1,
        "KAST": 67.5,
        "assists": 13,
        "deaths": 57,
        "firstKillsDifference": -7,
        "flashAssists": 3,
        "hsKills": 32,
        "killDeathsDifference": -5,
        "kills": 52,
        "player": {
          "id": 19206,
          "name": "jL",
        },
        "rating1": 0.96,
      },
      {
        "ADR": 60.8,
        "KAST": 66,
        "assists": 5,
        "deaths": 34,
        "firstKillsDifference": -6,
        "flashAssists": 0,
        "hsKills": 12,
        "killDeathsDifference": 0,
        "kills": 34,
        "player": {
          "id": 16885,
          "name": "Woro2k",
        },
        "rating1": 0.86,
      },
      {
        "ADR": 52.7,
        "KAST": 62.7,
        "assists": 14,
        "deaths": 62,
        "firstKillsDifference": -5,
        "flashAssists": 6,
        "hsKills": 17,
        "killDeathsDifference": -21,
        "kills": 41,
        "player": {
          "id": 14990,
          "name": "keen",
        },
        "rating1": 0.72,
      },
    ],
    "team2": [
      {
        "ADR": 87,
        "KAST": 74.7,
        "assists": 14,
        "deaths": 53,
        "firstKillsDifference": 4,
        "flashAssists": 2,
        "hsKills": 41,
        "killDeathsDifference": 17,
        "kills": 70,
        "player": {
          "id": 14932,
          "name": "faveN",
        },
        "rating1": 1.29,
      },
      {
        "ADR": 90.2,
        "KAST": 72.3,
        "assists": 20,
        "deaths": 60,
        "firstKillsDifference": 6,
        "flashAssists": 3,
        "hsKills": 35,
        "killDeathsDifference": 6,
        "kills": 66,
        "player": {
          "id": 7511,
          "name": "denis",
        },
        "rating1": 1.26,
      },
      {
        "ADR": 69.5,
        "KAST": 72.3,
        "assists": 15,
        "deaths": 51,
        "firstKillsDifference": 2,
        "flashAssists": 1,
        "hsKills": 21,
        "killDeathsDifference": 0,
        "kills": 51,
        "player": {
          "id": 12782,
          "name": "kressy",
        },
        "rating1": 1.05,
      },
      {
        "ADR": 67,
        "KAST": 72.3,
        "assists": 16,
        "deaths": 51,
        "firstKillsDifference": 0,
        "flashAssists": 6,
        "hsKills": 15,
        "killDeathsDifference": -2,
        "kills": 49,
        "player": {
          "id": 7499,
          "name": "Spiidi",
        },
        "rating1": 1.04,
      },
      {
        "ADR": 63.9,
        "KAST": 72.3,
        "assists": 24,
        "deaths": 51,
        "firstKillsDifference": 1,
        "flashAssists": 12,
        "hsKills": 16,
        "killDeathsDifference": 1,
        "kills": 52,
        "player": {
          "id": 15370,
          "name": "slaxz-",
        },
        "rating1": 0.99,
      },
    ],
  },
  "result": {
    "team1MapsWon": 1,
    "team2MapsWon": 2,
  },
  "team1": {
    "id": 8362,
    "name": "MAD Lions",
  },
  "team2": {
    "id": 8637,
    "name": "Sprout",
  },
}
`;
